commit 5db0a675b47ad5b2d6120383a93ef4ae3c0e4d00 (refs/changes/98/3698/2)
Author: Jordan Hendricks <jordan.hendricks@joyent.com>
Date:   2018-03-20T21:51:54+00:00 (1 year, 7 months ago)
    
    MANTA-3410 Audit job reports missing MPU parts
    MANTA-3608 update mola node-verror dependency

diff --git a/lib/auditor.js b/lib/auditor.js
index 7b592f9..40c8ed9 100644
--- a/lib/auditor.js
+++ b/lib/auditor.js
@@ -5,14 +5,29 @@
  */
 
 /*
- * Copyright (c) 2014, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
-var util = require('util');
-var events = require('events');
+var assert = require('assert-plus');
 var carrier = require('carrier');
+var events = require('events');
+var util = require('util');
 
 
+///--- Helpers
+
+/*
+ * Returns true if the given key is under the directory structure
+ * "/:accountUuid/uploads/", and false otherwise.
+ */
+function uploadsDirChild(key) {
+        assert.string(key, 'key');
+
+        /* JSSTYLED */
+        var uploadsRegex = /^\/[a-f0-9]{8}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{4}-[a-f0-9]{12}\/uploads\//;
+        return (key.match(uploadsRegex));
+}
+
 
 ///--- API
 
@@ -62,7 +77,20 @@ function Auditor(opts, listener) {
                 } else {
                         if ((currMako.objectId !== objectId) ||
                             (currMako.storageNodes.indexOf(storageId) === -1)) {
-                                self.emit('problem', line);
+
+                            var key = parts[3];
+
+                            /*
+                             * Committed, non-garbage collected multipart
+                             * uploads may have parts with no backing mako
+                             * file, which are removed during normal MPU
+                             * commit operations.
+                             */
+                            if (uploadsDirChild(key)) {
+                                    return;
+                            }
+
+                            self.emit('problem', line);
                         }
                 }
         });
diff --git a/package.json b/package.json
index 8d477f7..623d34b 100644
--- a/package.json
+++ b/package.json
@@ -23,7 +23,7 @@
                 "posix-getopt": "1.0.0",
                 "sprintf-js": "0.0.7",
                 "vasync": "1.6.1",
-                "verror": "^1.9.0",
+                "verror": "^1.10.0",
                 "vstream": "0.1.0"
         },
         "devDependencies": {
